libs/net/dal/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -118,4 +119,23 @@ public static IServiceCollection AddStorageConfig(this IServiceCollection servic
 
         return services;
     }
+
+    /// <summary>
+    /// Add the S3 configuration to the service collection.
+    /// </summary>
+    /// <param name="services"></param>
+    /// <param name="config"></param>
+    /// <returns></returns>
+    public static IServiceCollection AddS3Config(this IServiceCollection services, IConfiguration config)
+    {
+        services.Configure<S3Options>(config.GetSection("S3"));
```

Comment:
Dependency injection allows the implementation to control the configuration details.  This means it will populate the S3Options with configuration values found at the `S3` path. 

It is very odd that you are adding the configuration to memory and also creating a singleton.  Not only this, but you are creating a singleton from the copy in memory.

What are you attempting to achieve by this?  Is there a reason you need a singleton?  If so, why are you creating one this way?  Are you mixing configuration options with a helper class?